test/functional/p2p_rebroadcast.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+# Copyright (c) 2009-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test node rebroadcast logic.
+
+We start by creating a set of transactions to set the rebroadcast minimum fee
+cache to a medium fee rate value.
+
+We then create three sets of transactions:
+    1. aged high fee rate
+    2. aged low fee rate
+    3. recent high fee rate
+
+We add a new connection, trigger the rebroadcast functionality by mining an
+empty block, check that the aged high fee rate transactions were succesfully
+rebroadcast, and that the other two sets were not rebroadcast.
+"""
+
+from test_framework.p2p import P2PTxInvStore
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_approx,
+    assert_greater_than,
+    create_confirmed_utxos,
+)
+import time
+from decimal import Decimal
+
+# Constant from consensus.h
+MAX_BLOCK_WEIGHT = 4000000
+
+
+class NodeRebroadcastTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.extra_args = [[
+            "-whitelist=127.0.0.1",
+            "-txindex",
+            "-rebroadcast=1"
+        ]] * self.num_nodes
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def make_txn_at_fee_rate(self, input_utxo, outputs, outputs_sum, desired_fee_rate, change_address):
+        node = self.nodes[0]
+        node1 = self.nodes[1]
+
+        inputs = [{'txid': input_utxo['txid'], 'vout': input_utxo['vout']}]
+
+        # Calculate how much input values add up to
+        input_tx_hsh = input_utxo['txid']
+        raw_tx = node.decoderawtransaction(node.getrawtransaction(input_tx_hsh))
```

Comment:
In 7005767691005f66e7e08ec2e575d436dae3be47 "[test] Functional tests for rebroadcast logic."

`getrawtransaction` can decode txs too.


```suggestion
        raw_tx = node.getrawtransaction(input_tx_hsh, True)
```